### PR TITLE
rpmbuild: make sure detected epoch is int() or None

### PR DESCRIPTION
**Pull request by praiskup**
Resolves: #2848

### CODE REVIEW

File: backend/copr_backend/helpers.py [hunk 1]
```diff
@@ -634,12 +634,20 @@ def get_chroot_arch(chroot):
...
+    Return evr in format (epoch:)version-release.  The argument 'epoch' should
+    be integer value or null (but we rather consider "strings" here, too).
     """
-    if epoch is not None and epoch.isdigit():
-        return f"{epoch}:{version}-{release}"
+    if isinstance(epoch, str):
+        if epoch == "":
+            epoch = None
+        elif epoch.isdigit():
+            epoch = int(epoch)
```
Comment (FrostyX) [inline backend/copr_backend/helpers.py:644]: I think we don't need to convert the epoch to `int` here. The `return f"{epoch}:{version}-{release}"` value will be same in both cases. I think this function can remain as it was before? Maybe only with this small change to work for integers as well?

```diff
-    if epoch is not None and epoch.isdigit():
+    if epoch is not None and (isinstance(epoch, int) or epoch.isdigit()):
         return f"{epoch}:{version}-{release}"
```

File: rpmbuild/copr_rpmbuild/helpers.py [hunk 2]
```diff
@@ -406,7 +406,11 @@ def __init__(self, path):
         self.tags = self.spec.tags(self.spec.parsed_sections.package).content
 
     def __getattr__(self, name):
-        return getattr(self.spec, name)
+        value = getattr(self.spec, name)
+        if name == "epoch":
+            if value is not None:
```
Comment (FrostyX) [inline rpmbuild/copr_rpmbuild/helpers.py:411]: Nit: I'd go with

```python
if name == "epoch" and value is not None:
```

**Comment by FrostyX** [comment]
Thank you for the PR and sorry for introducing the 0 vs null inconsistency.

File: backend/copr_backend/helpers.py (hunk 1 above)
Comment (praiskup) [reply backend/copr_backend/helpers.py:644]: I had this variant, but it seemed harder to understand to me. But ... :-) I think I like it more now.

**Comment by praiskup** [comment]
> Thank you for the PR and sorry for introducing the 0 vs null inconsistency.

No problem! I'm sure I at least reviewed this before without complaints ;-) because it wasn't the first time I was thinking about the implicit Epoch value in RPM.

**Review by FrostyX — APPROVED** [review]

**Pull request merged**